File: VB/Models/DocumentsFactory.vb
Namespace DXRibbonNotepad.Model

    Public Module DocumentsFactory

        Public Function CreateDefaultDocuments() As Documents
            Dim result As Documents = CreateDocuments()
            result.Add(New Document())
            Return result
        End Function

        Public Function CreateDocuments() As Documents
            Return New Documents()
        End Function
    End Module
End Namespace

File: CS/Models/DocumentsFactory.cs
namespace DXRibbonNotepad.Model {
    public static class DocumentsFactory {
        public static Documents CreateDefaultDocuments() {
            Documents result = CreateDocuments();
            result.Add(new Document());
            return result;
        }
        public static Documents CreateDocuments() {
            return new Documents();
        }
    }
}